packages/elements/src/react/router/next.ts

```diff
@@ -21,7 +22,13 @@ export const useNextRouter = (): ClerkHostRouter => {
   return {
     mode: 'path',
     name: 'NextRouter',
-    push: (path: string) => router.push(path),
+    push: (path: string) => {
+      if (isAbsoluteUrl(path)) {
+        window.location.href = path;
+      } else {
+        router.push(path);
+      }
+    },
```

Comment:
Maybe i am missing some context here, but is this different from this [windowNavigate](https://github.com/clerk/javascript/blob/b6f0613dc9d8b0bab41cfabbaa8621b126e3bdf5/packages/clerk-js/src/utils/windowNavigate.ts#L11) within clerk-js

Reply:
Also I believe the nextjs router's `push` will respect the absolute path and it will perform the navigation.

Reply:
I totally thought the same thing about `push`, but in my testing it did not (at least with cross-origin domains, not sure about same-origin). I know the `<Link>` component handles cross-origin absolute URLs though, so I think that's where I got confused.

We can definitely bring `windowNavigate` over into Elements for the additional functionality. Good call!

Reply:
I tested with App router `router.push('https://google.com')` and seems to work. i've not tested with absolute url and same origin tho.

If you see something different, it may be an inconsistency across nextjs versions. so having the above logic will be required in that case.

Reply:
You're absolutely right. In Elements we have two routers, `ClerkRouter` and `FrameworkRouter`, which `ClerkRouter` wraps. I was calling `ClerkRouter.push(absoluteUrl)`, which doesn't work because it transforms the URL into `/?redirect_url=absoluteUrl`, and then calls `FrameworkRouter.push("/?redirect_url=absoluteUrl")`, which doesn't work. Calling `FrameworkRouter.push(absoluteUrl)` totally works as expected.

I'll close this PR in favor of a new one that reworks `ClerkRouter` to prevent the transformation of absolute URLs.